src/cmd/util_test.go

```diff
@@ -0,0 +1,82 @@
+package cmd
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+type Operation string
+
+const (
+	Assign   Operation = "assign"
+	Create   Operation = "create"
+	Delete   Operation = "delete"
+	Example  Operation = "example"
+	Get      Operation = "get"
+	List     Operation = "list"
+	Update   Operation = "update"
+	Unassign Operation = "unassign"
+)
+
+// execute any OpsLevel CLI command
+func execCmd(cmd Operation, resource string, inputs ...string) ([]byte, error) {
+	cliArgs := []string{string(cmd), resource}
+	cliArgs = append(cliArgs, inputs...)
+
+	r, oldStdout := redirectStdout()
+	defer r.Close()
+
+	rootCmd.SetArgs(cliArgs)
+	err := rootCmd.Execute()
+
+	output := captureOutput(r, oldStdout)
+	return output, err
+}
+
+// redirectStdout redirects os.Stdout to a pipe and returns the read and write ends of the pipe.
+func redirectStdout() (*os.File, *os.File) {
+	r, w, _ := os.Pipe()
+	oldStdout := os.Stdout
+	os.Stdout = w
+	return r, oldStdout
+}
+
+// captureOutput reads from r until EOF and returns the result as a string.
+func captureOutput(r *os.File, oldStdout *os.File) []byte {
+	w := os.Stdout
+	os.Stdout = oldStdout
+	w.Close()
+	gotOutput, _ := io.ReadAll(r)
+	return gotOutput
+	// return strings.TrimSpace(string(gotOutput))
+}
+
+// convert a simple API response to a string
+func asString(data []byte) string {
+	return strings.TrimSpace(string(data))
+}
+
+// convert JSON response from API to OpsLevel resource
+func jsonToResource[T any](jsonData []byte) *T {
```

Comment:
Also recommend that this function returns `(*T, error)` so that it can be handled by the calling test for example user_test.go line 37.